*Hess Antoine*
# Project 1

Vous êtes nouvel embauche chez CoolCorp, une startup qui spécialise
dans l'analyse des désastres maritimes.  Un nouveau client, White Star
Lines, se trouve face à un procès à travers le temps, affaire délicat.
Votre chef vous demande, en tant que seul data scientist chez
CoolCorp, d'analyser les données `*.csv`.  WSL a déjà engagé un expert
pour annoter une partie des données (`test.csv`) avec la survie ou pas
des passagers.

## Définition de l'intérêt de cette étude
Y a t-il une relation entre la survie des passagers et les différentes données les concernants ?

## Exploration des données
On commence par ouvrir le fichier test.csv avec la librairie **pandas** afin d'en obtenir un dataframe *dfTest* permettant de le manipuler :

In [17]:
import pandas as pd

#header = 0 pour indiquer la présence de titres pour les colonnes, se trouvant à la première ligne de notre fichier.
dfTrain = pd.read_csv('train.csv', header=0)

On peut commencer par afficher les premières lignes du dataframe, avec la méthode **head()** :

In [18]:
dfTrain.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

On obtient un premier aperçu des données et de leur organisation. Pour rappel :
- **PassengerId** : Le numéro d'embarquement du passager
- **Survived** : Indique si le passager a survécu (1) ou non (0)
- **Pclass** : La classe dans laquelle le passager était enregistré, 1, 2 ou 3.
- **Name** : Nom du passager
- **Sex** : Sexe du passager
- **Age** : Age du passager
- **SibSp** : Nombre de frères, soeurs épouse/époux du passager
- **Parch** : Nombre de pères/mères
- **Ticket** : Numéro du ticketdu passager
- **Fare** : Prix du ticker du passager
- **Cabin** : Numéro de cabine
- **Embarked** : Lieu d'embarcation



On peut ensuite obtenir quelques données intéréssantes grâce à la méthode **describe** :

In [20]:
dfTrain.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000         NaN    0.000000   
50%     446.000000    0.000000    3.000000         NaN    0.000000   
75%     668.500000    1.000000    3.000000         NaN    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

 - On peut déjà observer qu'il y a un total de 891 passagers et que nous avons bien des **PassengerId** commençant de 1 et se terminant à 891, à priori pas d'erreur (doublons ou manquants). 
 - La personne la plus agée a 80 ans et la moins agée moins de 1 an (l'âge des personnes ayant moins de 1 an sont représentées avec une valeur entre 0 et 1). L'âge moyen des passagers est de 29.67 ans. 
 - La valeur maximum de **SibSp** est de 8, indiquant qu'une personne a embarquée avec 8 frères/soeurs et épouse/époux. C'est une valeur élevée mais pas impossible.
 - En revanche, une valeure de 6 pour **Parch** indique qu'un passager à embarqué avec ses 6 pères/mères. Valeur abérente.
 - Le prix le plus élevé payé par un passager est de 512.3292. Cette somme peut paraitre étrange à cause du nombre de décimales ( des centimes de centimes de centimes ...) mais n'est pas abérante, on peut fortement imaginer qu'une suite de luxe se trouve sur le bâteau.
 - On remarque que l'âge de tous les passagers n'a pas été renseigné.

On peut ensuite afficher un petit résumé plus complet sur les données, avec notamment le type des variables grâce à la méthode **info()** :

In [21]:
dfTrain.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
PassengerId    891 non-null int64
Survived       891 non-null int64
Pclass         891 non-null int64
Name           891 non-null object
Sex            891 non-null object
Age            714 non-null float64
SibSp          891 non-null int64
Parch          891 non-null int64
Ticket         891 non-null object
Fare           891 non-null float64
Cabin          204 non-null object
Embarked       889 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 83.6+ KB


## Néttoyage des données 